packages/core-data/src/actions.js

```diff
@@ -256,7 +255,9 @@ export const deleteEntityRecord =
 	) =>
 	async ( { dispatch } ) => {
 		const configs = await dispatch( getOrLoadEntitiesConfig( kind ) );
-		const entityConfig = find( configs, { kind, name } );
+		const entityConfig = configs.find(
```

Comment:
`configs` will always be an array:

https://github.com/WordPress/gutenberg/blob/a4ce174b04ed471360b35482d80a453286d7d2c5/packages/core-data/src/entities.js#L322